### PR TITLE
Library reorganisation

### PR DESCRIPTION
**Pull request by james-ward**
Move files around to make this function as a library

### CODE REVIEW

**Review by ArthurAllshire — APPROVED** [review]
There are a couple of issues with some of the functions that were implemented in the controller but I think its best to merge them and write tests to finalise them. The reorganization at this point looks good to me.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+      description='FRC Team 4774\'s Swerve Drive library',
+      url='http://github.com/thedropbears/libswervedrive',
+      author='FRC Team 4774',
+      author_email='enquiries@thedropbears.org.au',
+      license='MIT',
+      packages=['swervedrive'],
+      install_requires=[
+          'numpy',
+      ],
+      zip_safe=False)
```
Comment (auscompgeek) [inline setup.py]: I can't see any evidence that this is actually correct. Is it?

Comment (james-ward) [reply setup.py]: Should it be true, or omitted completely?

Comment (auscompgeek) [reply setup.py]: If you omit this, setuptools will try to determine the correct value automatically.

Comment (auscompgeek) [reply setup.py]: This is also a legacy thing anyway, dating back to the days of egg distributions...

**Pull request merged**